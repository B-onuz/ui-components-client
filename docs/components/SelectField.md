# SelectField (Component)

Componente para campo de seleção em formulários.

## Como utilizar

```js
import SelectField from 'ui-components/lib/components/SelectField'
```

```jsx
import React, { useState } from 'react'
import SelectField from 'ui-components/lib/components/SelectField'
import { Box, Flex } from 'reflexbox'

const Controlled = () => {
  const [selected, setSelected] = useState('valor4')
  const handleChangeSelect = (e) => {
    setSelected(e.target.value)
  }
  return (
    <Flex flexWrap={'wrap'}>
      <Box width={[1, 1 / 2, 1 / 3]}>
        <SelectField value={selected} onChange={handleChangeSelect} id="name" name={'name'} required label={'Nome'} m={2} placeholder={'Nome'}>
          <option value="valor1">Valor 1</option>
          <option value="valor2">Valor 2</option>
          <option value="valor3">Valor 3</option>
          <option value="valor4">Valor 4</option>
          <option value="valor5">Valor 5</option>
          <option value="valor6">Valor 6</option>
          <option value="valor7">Valor 7</option>
          <option value="valor8">Valor 8</option>
        </SelectField>
      </Box>
    </Flex>
  )
}
export default Controlled
```
