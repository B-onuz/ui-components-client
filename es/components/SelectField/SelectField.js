"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Label = _interopRequireDefault(require("../Label"));

var _Select = _interopRequireDefault(require("../Select"));

var _FormHelperText = _interopRequireDefault(require("../FormHelperText"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledSystem = require("styled-system");

var _propTypes2 = _interopRequireDefault(require("@styled-system/prop-types"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/SelectField/SelectField.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  display: block;\n  font-family: inherit;\n  ", "\n  label {\n    display: block;\n    margin-bottom: 0.4em;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledSelectField = _styledComponents["default"].div(_templateObject(), (0, _styledSystem.compose)(_styledSystem.space, _styledSystem.layout));

var SelectField = function SelectField(_ref) {
  var children = _ref.children,
      label = _ref.label,
      placeholder = _ref.placeholder,
      name = _ref.name,
      id = _ref.id,
      value = _ref.value,
      onChange = _ref.onChange,
      required = _ref.required,
      error = _ref.error,
      helperText = _ref.helperText,
      _ref$inputProps = _ref.inputProps,
      inputProps = _ref$inputProps === void 0 ? {} : _ref$inputProps,
      rest = _objectWithoutPropertiesLoose(_ref, ["children", "label", "placeholder", "name", "id", "value", "onChange", "required", "error", "helperText", "inputProps"]);

  return /*#__PURE__*/_react["default"].createElement(StyledSelectField, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 22,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react["default"].createElement(_Label["default"], {
    htmlFor: id,
    color: !!error && 'danger',
    required: !!required,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 23,
      columnNumber: 7
    }
  }, label), /*#__PURE__*/_react["default"].createElement(_Select["default"], _extends({
    id: id,
    name: name,
    placeholder: placeholder
    /* controlled props */
    ,
    value: value,
    onChange: onChange
    /* validation props */
    ,
    required: !!required,
    "aria-required": !!required,
    "aria-invalid": !!error,
    error: error
  }, inputProps, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 26,
      columnNumber: 7
    }
  }), children), !!error && /*#__PURE__*/_react["default"].createElement(_FormHelperText["default"], {
    color: 'red',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 42,
      columnNumber: 19
    }
  }, error), !!helperText && /*#__PURE__*/_react["default"].createElement(_FormHelperText["default"], {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 43,
      columnNumber: 24
    }
  }, helperText));
};

SelectField.propTypes = process.env.NODE_ENV !== "production" ? _extends({}, _propTypes2["default"].space, {}, _propTypes2["default"].layout, {
  multiline: _propTypes["default"].bool
}) : {};
SelectField.defaultProps = {
  required: false,
  error: false,
  multiline: false
};
var _default = SelectField;
exports["default"] = _default;