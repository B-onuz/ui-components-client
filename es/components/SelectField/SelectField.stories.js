'use strict'

function _typeof(obj) {
  '@babel/helpers - typeof'
  if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
    _typeof = function _typeof(obj) {
      return typeof obj
    }
  } else {
    _typeof = function _typeof(obj) {
      return obj && typeof Symbol === 'function' && obj.constructor === Symbol && obj !== Symbol.prototype ? 'symbol' : typeof obj
    }
  }
  return _typeof(obj)
}

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.HelperText = exports.ErrorMessage = exports.Controlled = exports['default'] = void 0

var _react = _interopRequireWildcard(require('react'))

var _SelectField = _interopRequireDefault(require('./SelectField'))

var _reflexbox = require('reflexbox')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/SelectField/SelectField.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

function _getRequireWildcardCache() {
  if (typeof WeakMap !== 'function') return null
  var cache = new WeakMap()
  _getRequireWildcardCache = function _getRequireWildcardCache() {
    return cache
  }
  return cache
}

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj
  }
  if (obj === null || (_typeof(obj) !== 'object' && typeof obj !== 'function')) {
    return { default: obj }
  }
  var cache = _getRequireWildcardCache()
  if (cache && cache.has(obj)) {
    return cache.get(obj)
  }
  var newObj = {}
  var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor
  for (var key in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null
      if (desc && (desc.get || desc.set)) {
        Object.defineProperty(newObj, key, desc)
      } else {
        newObj[key] = obj[key]
      }
    }
  }
  newObj['default'] = obj
  if (cache) {
    cache.set(obj, newObj)
  }
  return newObj
}

var _default = {
  title: 'SelectField',
  component: _SelectField['default'],
}
exports['default'] = _default

var Controlled = function Controlled() {
  var _useState = (0, _react.useState)('valor4'),
    selected = _useState[0],
    setSelected = _useState[1]

  var handleChangeSelect = function handleChangeSelect(e) {
    setSelected(e.target.value)
  }

  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      flexWrap: 'wrap',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 16,
        columnNumber: 5,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        width: [1, 1 / 2, 1 / 3],
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 17,
          columnNumber: 7,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _SelectField['default'],
        {
          value: selected,
          onChange: handleChangeSelect,
          id: 'name',
          name: 'name',
          required: true,
          label: 'Nome',
          m: 2,
          placeholder: 'Nome',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 18,
            columnNumber: 9,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor1',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 19,
              columnNumber: 11,
            },
          },
          'Valor 1'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor2',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 20,
              columnNumber: 11,
            },
          },
          'Valor 2'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor3',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 21,
              columnNumber: 11,
            },
          },
          'Valor 3'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor4',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 22,
              columnNumber: 11,
            },
          },
          'Valor 4'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor5',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 23,
              columnNumber: 11,
            },
          },
          'Valor 5'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor6',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 24,
              columnNumber: 11,
            },
          },
          'Valor 6'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor7',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 25,
              columnNumber: 11,
            },
          },
          'Valor 7'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor8',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 26,
              columnNumber: 11,
            },
          },
          'Valor 8'
        )
      )
    )
  )
}

exports.Controlled = Controlled

var ErrorMessage = function ErrorMessage() {
  var _useState2 = (0, _react.useState)('valor4'),
    selected = _useState2[0],
    setSelected = _useState2[1]

  var handleChangeSelect = function handleChangeSelect(e) {
    setSelected(e.target.value)
  }

  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      flexWrap: 'wrap',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 39,
        columnNumber: 5,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        width: [1, 1 / 2, 1 / 3],
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 40,
          columnNumber: 7,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _SelectField['default'],
        {
          value: selected,
          error: 'Error message',
          onChange: handleChangeSelect,
          id: 'name',
          name: 'name',
          required: true,
          label: 'Nome',
          m: 2,
          placeholder: 'Nome',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 41,
            columnNumber: 9,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor1',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 52,
              columnNumber: 11,
            },
          },
          'Valor 1'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor2',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 53,
              columnNumber: 11,
            },
          },
          'Valor 2'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor3',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 54,
              columnNumber: 11,
            },
          },
          'Valor 3'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor4',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 55,
              columnNumber: 11,
            },
          },
          'Valor 4'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor5',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 56,
              columnNumber: 11,
            },
          },
          'Valor 5'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor6',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 57,
              columnNumber: 11,
            },
          },
          'Valor 6'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor7',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 58,
              columnNumber: 11,
            },
          },
          'Valor 7'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor8',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 59,
              columnNumber: 11,
            },
          },
          'Valor 8'
        )
      )
    )
  )
}

exports.ErrorMessage = ErrorMessage

var HelperText = function HelperText() {
  var _useState3 = (0, _react.useState)('valor4'),
    selected = _useState3[0],
    setSelected = _useState3[1]

  var handleChangeSelect = function handleChangeSelect(e) {
    setSelected(e.target.value)
  }

  return /*#__PURE__*/ _react['default'].createElement(
    _reflexbox.Flex,
    {
      flexWrap: 'wrap',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 72,
        columnNumber: 5,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Box,
      {
        width: [1, 1 / 2, 1 / 3],
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 73,
          columnNumber: 7,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _SelectField['default'],
        {
          value: selected,
          helperText: 'Helper text',
          onChange: handleChangeSelect,
          id: 'name',
          name: 'name',
          required: true,
          label: 'Nome',
          m: 2,
          placeholder: 'Nome',
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 74,
            columnNumber: 9,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor1',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 85,
              columnNumber: 11,
            },
          },
          'Valor 1'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor2',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 86,
              columnNumber: 11,
            },
          },
          'Valor 2'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor3',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 87,
              columnNumber: 11,
            },
          },
          'Valor 3'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor4',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 88,
              columnNumber: 11,
            },
          },
          'Valor 4'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor5',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 89,
              columnNumber: 11,
            },
          },
          'Valor 5'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor6',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 90,
              columnNumber: 11,
            },
          },
          'Valor 6'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor7',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 91,
              columnNumber: 11,
            },
          },
          'Valor 7'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          'option',
          {
            value: 'valor8',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 92,
              columnNumber: 11,
            },
          },
          'Valor 8'
        )
      )
    )
  )
}

exports.HelperText = HelperText
