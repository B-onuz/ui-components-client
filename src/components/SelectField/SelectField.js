import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Select from '../Select'
import FormHelperText from '../FormHelperText'
import PropTypes from 'prop-types'
import { space, layout, compose, flexbox } from 'styled-system'
import propTypes from '@styled-system/prop-types'

const StyledSelectField = styled.div`
  display: block;
  font-family: inherit;
  ${compose(space, layout)}
  label {
    display: block;
    margin-bottom: 0.4em;
  }
`

const SelectField = ({ children, label, placeholder, name, id, value, onChange, required, error, helperText, inputProps = {}, ...rest }) => {
  return (
    <StyledSelectField {...rest}>
      <Label htmlFor={id} color={!!error && 'danger'} required={!!required}>
        {label}
      </Label>
      <Select
        id={id}
        name={name}
        placeholder={placeholder}
        /* controlled props */
        value={value}
        onChange={onChange}
        /* validation props */
        required={!!required}
        aria-required={!!required}
        aria-invalid={!!error}
        error={error}
        {...inputProps}
      >
        {children}
      </Select>
      {!!error && <FormHelperText color={'red'}>{error}</FormHelperText>}
      {!!helperText && <FormHelperText>{helperText}</FormHelperText>}
    </StyledSelectField>
  )
}

SelectField.propTypes = {
  ...propTypes.space,
  ...propTypes.layout,
  multiline: PropTypes.bool,
}

SelectField.defaultProps = {
  required: false,
  error: false,
  multiline: false,
}

export default SelectField
