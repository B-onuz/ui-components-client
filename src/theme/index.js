const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
