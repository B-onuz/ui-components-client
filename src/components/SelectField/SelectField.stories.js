import React, { useState } from 'react'
import SelectField from './SelectField'
import { Box, Flex } from 'reflexbox'

export default {
  title: 'SelectField',
  component: SelectField,
}

export const Controlled = () => {
  const [selected, setSelected] = useState('valor4')
  const handleChangeSelect = (e) => {
    setSelected(e.target.value)
  }
  return (
    <Flex flexWrap={'wrap'}>
      <Box width={[1, 1 / 2, 1 / 3]}>
        <SelectField value={selected} onChange={handleChangeSelect} id="name" name={'name'} required label={'Nome'} m={2} placeholder={'Nome'}>
          <option value="valor1">Valor 1</option>
          <option value="valor2">Valor 2</option>
          <option value="valor3">Valor 3</option>
          <option value="valor4">Valor 4</option>
          <option value="valor5">Valor 5</option>
          <option value="valor6">Valor 6</option>
          <option value="valor7">Valor 7</option>
          <option value="valor8">Valor 8</option>
        </SelectField>
      </Box>
    </Flex>
  )
}

export const ErrorMessage = () => {
  const [selected, setSelected] = useState('valor4')
  const handleChangeSelect = (e) => {
    setSelected(e.target.value)
  }
  return (
    <Flex flexWrap={'wrap'}>
      <Box width={[1, 1 / 2, 1 / 3]}>
        <SelectField
          value={selected}
          error={'Error message'}
          onChange={handleChangeSelect}
          id="name"
          name={'name'}
          required
          label={'Nome'}
          m={2}
          placeholder={'Nome'}
        >
          <option value="valor1">Valor 1</option>
          <option value="valor2">Valor 2</option>
          <option value="valor3">Valor 3</option>
          <option value="valor4">Valor 4</option>
          <option value="valor5">Valor 5</option>
          <option value="valor6">Valor 6</option>
          <option value="valor7">Valor 7</option>
          <option value="valor8">Valor 8</option>
        </SelectField>
      </Box>
    </Flex>
  )
}

export const HelperText = () => {
  const [selected, setSelected] = useState('valor4')
  const handleChangeSelect = (e) => {
    setSelected(e.target.value)
  }
  return (
    <Flex flexWrap={'wrap'}>
      <Box width={[1, 1 / 2, 1 / 3]}>
        <SelectField
          value={selected}
          helperText={'Helper text'}
          onChange={handleChangeSelect}
          id="name"
          name={'name'}
          required
          label={'Nome'}
          m={2}
          placeholder={'Nome'}
        >
          <option value="valor1">Valor 1</option>
          <option value="valor2">Valor 2</option>
          <option value="valor3">Valor 3</option>
          <option value="valor4">Valor 4</option>
          <option value="valor5">Valor 5</option>
          <option value="valor6">Valor 6</option>
          <option value="valor7">Valor 7</option>
          <option value="valor8">Valor 8</option>
        </SelectField>
      </Box>
    </Flex>
  )
}
