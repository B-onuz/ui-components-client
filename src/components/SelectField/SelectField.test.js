import React from 'react'
import 'jest-styled-components'
import SelectField from './SelectField.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('SelectField changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <SelectField id="name" name={'name'} required label={'Nome'} m={2} placeholder={'Nome'} />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
