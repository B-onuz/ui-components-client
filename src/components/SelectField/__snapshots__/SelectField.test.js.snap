// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField changes the class when hovered 1`] = `
.c1 {
  color: #555;
  font-size: 0.9em;
  font-weight: 600;
}

.c2 {
  border: 1px solid #dededf;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  line-height: inherit;
  border-radius: 4px;
  line-height: 2.1em;
  outline: none;
  padding-right: 1em;
  padding-left: 1em;
  width: 100%;
  box-sizing: border-box;
  background: #fff;
  background-image: url('data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTc2cHgiIGhlaWdodD0iOTBweCIgdmlld0JveD0iMCAwIDE3NiA5MCIgdmVyc2lvbj0iMS4xIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4KICAgIDx0aXRsZT5UcmlhbmdsZTwvdGl0bGU+CiAgICA8ZyBpZD0iV2VsY29tZSIgc3Ryb2tlPSJub25lIiBzdHJva2Utd2lkdGg9IjEiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+CiAgICAgICAgPGcgaWQ9IkFydGJvYXJkIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgtMTYuMDAwMDAwLCAtNy4wMDAwMDApIiBmaWxsPSIjNEQ0RDREIj4KICAgICAgICAgICAgPHBhdGggZD0iTTEwNS40MTQyMTQsOC40MTQyMTM1NiBMMTkwLjU4NTc4Niw5My41ODU3ODY0IEMxOTEuMzY2ODM1LDk0LjM2NjgzNSAxOTEuMzY2ODM1LDk1LjYzMzE2NSAxOTAuNTg1Nzg2LDk2LjQxNDIxMzYgQzE5MC4yMTA3MTQsOTYuNzg5Mjg2MyAxODkuNzAyMDA2LDk3IDE4OS4xNzE1NzMsOTcgTDE4LjgyODQyNzEsOTcgQzE3LjcyMzg1NzYsOTcgMTYuODI4NDI3MSw5Ni4xMDQ1Njk1IDE2LjgyODQyNzEsOTUgQzE2LjgyODQyNzEsOTQuNDY5NTY3IDE3LjAzOTE0MDgsOTMuOTYwODU5MiAxNy40MTQyMTM2LDkzLjU4NTc4NjQgTDEwMi41ODU3ODYsOC40MTQyMTM1NiBDMTAzLjM2NjgzNSw3LjYzMzE2NDk4IDEwNC42MzMxNjUsNy42MzMxNjQ5OCAxMDUuNDE0MjE0LDguNDE0MjEzNTYgWiIgaWQ9IlRyaWFuZ2xlIiB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxMDQuMDAwMDAwLCA1Mi4wMDAwMDApIHJvdGF0ZSgtMTgwLjAwMDAwMCkgdHJhbnNsYXRlKC0xMDQuMDAwMDAwLCAtNTIuMDAwMDAwKSAiPjwvcGF0aD4KICAgICAgICA8L2c+CiAgICA8L2c+Cjwvc3ZnPg==');
  background-repeat: no-repeat,repeat;
  cursor: pointer;
  background-position: right 0.7em top 50%,0 0;
  background-size: 0.65em auto,100%;
  font-size: 16px;
}

.c2::-webkit-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c2::-moz-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c2:-ms-input-placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c2::placeholder {
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  color: #b5b5b5;
}

.c2:focus {
  box-shadow: 0 0 3px rgba(33,133,208,0.5);
  border-color: #2185d0;
}

.c0 {
  display: block;
  font-family: inherit;
  margin: 8px;
}

.c0 label {
  display: block;
  margin-bottom: 0.4em;
}

<div
  className="c0"
>
  <label
    className="c1"
    color={false}
    htmlFor="name"
  >
    <abbr
      title="Campo obrigatÃ³rio"
    >
      *
    </abbr>
     Nome
  </label>
  <select
    aria-invalid={false}
    aria-required={true}
    className="c2"
    color="blue"
    id="name"
    name="name"
    placeholder="Nome"
    required={true}
    size="medium"
  />
</div>
`;
